code/controllers/ModelAsController.php

```diff
@@ -49,82 +49,79 @@ public function init() {
 	public function handleRequest(SS_HTTPRequest $request, DataModel $model) {
 		$this->setRequest($request);
 		$this->setDataModel($model);
-		
-		$this->pushCurrent();
 
-		// Create a response just in case init() decides to redirect
-		$this->response = new SS_HTTPResponse();
```

Comment:
Sorry for the delay, I think storing a local `$response = $this->getResponse()` variable before `init()` is called should cover this, right? Because then `$this->response` will be set